File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from app.schemas.common import PaginationFields


class NotificationResponse(BaseModel):
    id: str
    event_type: str
    category: str
    severity: Literal["info", "success", "warning", "critical"]
    title: str
    message: str
    action_url: str | None
    metadata: dict
    is_read: bool
    created_at: datetime


class NotificationListResponse(PaginationFields):
    items: list[NotificationResponse]
    unread_count: int


class NotificationUnreadResponse(BaseModel):
    unread_count: int


class NotificationPreferences(BaseModel):
    in_app_enabled: bool = True
    email_enabled: bool = False
    security_alerts: bool = True
    qr_activity: bool = True
    share_activity: bool = True
    bulk_activity: bool = True
    local_smtp_available: bool = False


class NotificationPreferencesUpdate(BaseModel):
    in_app_enabled: bool
    email_enabled: bool
    security_alerts: bool
    qr_activity: bool
    share_activity: bool
    bulk_activity: bool

File: frontend/src/features/notifications/types/notification.ts
import type { PaginatedResponse } from "../../../lib/api/types";

export type NotificationSeverity = "info" | "success" | "warning" | "critical";
export interface NotificationItem { id: string; event_type: string; category: string; severity: NotificationSeverity;
  title: string; message: string; action_url: string | null; metadata: Record<string, unknown>; is_read: boolean; created_at: string; }
export interface NotificationList extends PaginatedResponse<NotificationItem> { unread_count: number; }
export interface NotificationPreferences { in_app_enabled: boolean; email_enabled: boolean; security_alerts: boolean;
  qr_activity: boolean; share_activity: boolean; bulk_activity: boolean; local_smtp_available: boolean; }
export type NotificationPreferencesUpdate = Omit<NotificationPreferences, "local_smtp_available">;
